src/persist-client/src/internal/metrics.rs

```diff
@@ -1654,6 +1670,185 @@ impl SinkMetrics {
+        // in each worker. So upon quiescence, the net effect of these must result in a
+        // non-negative quantity.
+        // 2. Every worker will execute the instructions below and compute the same peak.
+        // This is because during quiescence, no changes happen to the previous peak nor
+        // other metrics, and the ones we look at are all shared across workers. So at least
+        // one worker will update the peak with its new value, if there is an increase, at
+        // the quiescence granularity.
+        // Note that we may miss the overall peak if between quiescence moments there are
+        // lots of insertions followed by lots of deletions to the correction buffer. We
+        // find this trade-off more attractive than adding explicity multi-metric synchronization
+        // among workers.
+        let aggregate_correction_len = self
+            .correction_sum_insertions
+            .get()
+            .checked_sub(self.correction_sum_deletions.get());
+        let Some(aggregate_correction_len) = aggregate_correction_len else {
+            error!(
+                aggregate_correction_len,
+                "Negative aggregate length for persist sink correction"
+            );
+            return;
+        };
+        if aggregate_correction_len > self.correction_peak_len.get() {
+            self.correction_peak_len.set(aggregate_correction_len);
+        }
+        let aggregate_correction_cap = self
+            .correction_sum_capacity_increases
+            .get()
+            .checked_sub(self.correction_sum_capacity_decreases.get());
+        let Some(aggregate_correction_cap) = aggregate_correction_cap else {
+            error!(
+                aggregate_correction_cap,
+                "Negative aggregate capacity for persist sink correction"
+            );
+            return;
+        };
+        if aggregate_correction_cap > self.correction_peak_capacity.get() {
+            self.correction_peak_capacity.set(aggregate_correction_cap);
+        }
+    }
+}
+
+/// Metrics for the persist sink that are labeled per-worker.
+#[derive(Clone, Debug)]
+pub struct SinkWorkerMetrics {
+    correction_sum_insertions: UIntGauge,
+    correction_sum_deletions: UIntGauge,
+    correction_previous_len: usize,
+    correction_max_len_per_worker: UIntGauge,
+    correction_sum_capacity_increases: UIntGauge,
+    correction_sum_capacity_decreases: UIntGauge,
+    correction_previous_cap: usize,
+    correction_max_capacity_per_worker: UIntGauge,
+}
+
+impl SinkWorkerMetrics {
+    pub fn update_correction_metrics(&mut self, correction_len: usize, correction_cap: usize) {
+        if correction_len > self.correction_previous_len {
+            // Report insertions.
+            let insertions = correction_len - self.correction_previous_len;
+            self.correction_sum_insertions.add(
+                insertions
+                    .try_into()
```

Comment:
`CastFrom::cast_from` for errorless conversion of usize to u64, here and below.

Reply:
Nice, done!